### ISSUE TITLE
How do I set WS connection headers?

### ISSUE THREAD

**Issue by raaaar** [Azure/azure-c-shared-utility#140]
I need to specify a few headers before establishing a WS connection and cannot quite figure out how to do that using uws_client.

Please, help!

**Comment by ewertons** [comment]
Hey @raaaar ,

if you inspect this repo you will see that the headers are saved in a directory with the same name under inc:
https://github.com/Azure/azure-c-shared-utility/tree/master/inc/azure_c_shared_utility

So if you want to include headers from uws_client you need to use:

```
#include "azure_c_shared_utility/wsio.h"
```

See example [here](https://github.com/Azure/azure-iot-sdk-c/blob/master/iothub_client/src/iothubtransportamqp_websockets.c#L5).

**Comment by ewertons** [comment]
You can step into wsio module above to see how to include the other websocket headers from c-shared-utility lib.

**Comment by raaaar** [comment]
That's not exactly what I was asking about. Should've been more specific, sorry. 

Let me rephrase: in the HTTP Upgrade I need to send a few extra HTTP headers (authorization token, connection id, etc.; take a look at the very first HTTP request on [this page](https://docs.microsoft.com/en-us/azure/cognitive-services/speech/api-reference-rest/websocketprotocol)). So, I was wondering if this is currently supported and, if not, whether you have any plans to add this functionality?

**Comment by raaaar** [comment]
Any updates?

**Comment by raaaar** [comment]
+ @zhouwangzw

**Comment by ewertons** [comment]
Hi @raaaar ,
no problem.

No we do not expose an option to set headers.
Unfortunately the development in this library is guided towards the needs to the azure-iot-sdk-c repo.

However, if you would like to give it a try and submit a PR we will work together to get it approved and merged.

**Comment by jspaith** [comment]
Since this has been dormant so long, I'm going to close out.  Feel free to reactivate or open a new issue if needed.

**Comment by ewertons** [comment]
Hi @raaaar ,
just to share the details of this improvement with you, we added the capability of setting request headers in the uws_client module.

To use it just call this function after creating the uws_client:
```c
int result = uws_client_set_request_header(uws_client_handle, "HeaderName", "Header Value");
```
You can check the new function [here](https://github.com/Azure/azure-c-shared-utility/blob/28e6dda07a07b2daa934917153f3619c81ee7e14/inc/azure_c_shared_utility/uws_client.h#L110).

Thanks,
Azure IoT SDK Team.
